JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -912,6 +934,16 @@ - (void)keyboardController:(JSQMessagesKeyboardController *)keyboardController k
 
     heightFromBottom = MAX(0.0, heightFromBottom);
 
+    // Hide the picker view if the keyboard/input view is dismissed via the
+    // pan gesture recognizer
+
+    if (heightFromBottom == 0.0 && self.isPickerViewVisible) {
+        self.isPickerViewVisible = NO;
+    }
+
+    // Show/hide the ad container view
+    self.isAdVisible = heightFromBottom == 0.0;
```

Comment:
This determines if we should show or hide the ad container based on whether or not the keyboard or picker view is showing.